### PR TITLE
Feature/advanced plotting/contour plotting [5/6]

### PR DESCRIPTION
**Pull request by DJAntivenom**
# Contour Plotting

This PR adds the code needed for contour plotting.

### CODE REVIEW

File: LabExT/Experiments/StandardExperiment.py [hunk 1]
```diff
@@ -409,6 +409,48 @@ def _write_metadata(self, target: dict = None, file_path: str = "tmp.json") -> d
...
+        3. Iterate through the "keys" of the dict-like object
+            a. If the corresponding value is dict-like itself, try to get the value corresponding to the key "value".
+            b. Otherwise assume the value is some sort of float/int/bool/... and use it directly
+
+        Args:
+            meas_dict: The meas_dict which is used to guess the used parameters.
+
+        Returns:
+            A new dict containing the guessed name-value pairs.
+        """
```
Comment (marcoep) [inline LabExT/Experiments/StandardExperiment.py:428]: This function should become superfluous once the requested changes in #199 are implemented. Please do not use two different keys to store the same measurement parameters.

File: LabExT/Experiments/StandardExperiment.py [hunk 2]
```diff
@@ -460,6 +502,12 @@ def load_measurement_dataset(self, meas_dict, file_path, force_gui_update=True):
         if meas_hash in self.measurements_hashes:
             raise ValueError("Duplicate measurement found!")
 
+        if "sweep_information" not in meas_dict.keys():
+            meas_dict["sweep_information"] = {"part_of_sweep": False, "sweep_association": dict()}
+
+        if "measurement_params" not in meas_dict.keys():
+            meas_dict["measurement_params"] = self.__guess_params(meas_dict)
```
Comment (marcoep) [inline LabExT/Experiments/StandardExperiment.py:509]: same as above, please continue using the existing "measurement settings" key.

File: LabExT/View/Controls/Plotting/PlotController.py [hunk 3]
```diff
@@ -104,11 +108,149 @@ def __draw_axes_line_plot(self, axis_x_key: str, axis_y_key: str) -> None:
...
+        if len(self._plottable_data) < 2:
+            # for a contour plot we need at least 2 selected measurements
+            return
+
+        if "" in [
+            self._model.axis_x_key_name.get(),
+            self._model.axis_y_key_name.get(),
+            self._model.axis_z_key_name.get(),
+        ]:
+            # the variables are set to "" by the settings frame if something doesn't work
```
Comment (marcoep) [inline LabExT/View/Controls/Plotting/PlotController.py:122]: can you be more specific? What's happening if "something doesnt work"?

**Review by marcoep — CHANGES_REQUESTED** [review]

File: LabExT/Experiments/StandardExperiment.py (hunk 1 above)
Comment (DJAntivenom) [reply LabExT/Experiments/StandardExperiment.py]: That's correct. Removed in 73b76d749fac742ecbb6c31cd6045c68d7107887.

File: LabExT/Experiments/StandardExperiment.py (hunk 2 above)
Comment (DJAntivenom) [reply LabExT/Experiments/StandardExperiment.py]: Done in 73b76d749fac742ecbb6c31cd6045c68d7107887.

File: LabExT/View/Controls/Plotting/PlotController.py (hunk 3 above)
Comment (DJAntivenom) [reply LabExT/View/Controls/Plotting/PlotController.py]: Added more information in 1a3cf748c9c7a0dcfa5e5cc136615e2406d647a8.

**Review by marcoep — APPROVED** [review]

**Comment by DJAntivenom** [comment]
Superseded by #211

**Pull request closed without merging**